redis/src/cluster_async/mod.rs

```diff
@@ -1210,6 +1219,9 @@ where
             if Self::check_for_topology_diff(inner.clone()).await {
                 let _ = Self::refresh_slots_and_subscriptions_with_retries(inner.clone()).await;
             } else {
+                // This serves as a safety measure for validating pubsub subsctiptions state in case it has drifted
+                // while topology stayed the same. Currently it should not happen, but the cost of missing some corner cases
```

Comment:
We now know when it can happen - if the topology haven't changed but the connection was removed from the connection map, calling refresh_pubsub_subscriptions will try to refresh it

Reply:
fixed